core/store/mmap_directory.cpp

```diff
@@ -244,9 +242,9 @@ index_input::ptr CachingMMapDirectory::open(std::string_view name,
     return make_stream(std::move(handle));
   }
 
-  if (handle = OpenHandle(directory(), name, advice,
-                          *resource_manager_.file_descriptors);
-      handle) {
```

Comment:
Make it more difficult for reader, for compilation, for debugging, but without profit like smaller count of lines or more granular scope